lib/services/nodes-api-service.js

```diff
@@ -451,5 +453,80 @@ function nodeApiServiceFactory(
             });
     };
 
+    /**
+     * Add the tags to the specified node
+     * @param  {String}     id
+     * @param  {Array}      tags
+     * @return {Promise}
+     */
+    NodeApiService.prototype.addTagsById = function(id, tags) {
+        return Promise.resolve().then(function() {
+            assert.ok(Array.isArray(tags), 'tags must be an array');
+            assert.isMongoId(id, 'the id must be a valid mongo id');
+        })
+        .then(function() {
+            return waterline.nodes.needByIdentifier(id);
+        })
+        .then(function () {
+            return waterline.nodes.addTags(id, tags);
+        })
+        .then(function() {
+            return tags;
+        });
+    };
+
+    /**
+     * Remove the tag from the specified node
+     * @param  {String}     id
+     * @param  {String}     tagName
+     * @return {Promise}
+     */
+    NodeApiService.prototype.removeTagsById = function(id, tagName) {
+        return Promise.resolve().then(function() {
+            assert.string(tagName, 'tag must be a string');
+            assert.isMongoId(id, 'the id must be a valid mongo id');
+        })
+        .then(function() {
+            return waterline.nodes.needByIdentifier(id);
+        })
+        .then(function () {
+            return waterline.nodes.remTags(id, tagName);
+        })
+        .then(function() {
+            return tagName;
+        });
+    };
+
+    /**
+     * Get a list of tags applied to the specified id
+     * @param  {String}     id
+     * @return {Promise}    Resolves to an array of tags
+     */
+    NodeApiService.prototype.getTagsById = function(id) {
+        return Promise.resolve().then(function() {
```

Comment:
@zyoung51 I mean if error handing is already in an area of concentration, the throwed assert exception in each function could be handled there,  so don't need to be promised in each function here. and if don't want complexity of try/catch in somewhere else, promise's catch also could work there.